docs/proposals/runtime-client-server.md

```diff
@@ -0,0 +1,351 @@
+    // The spec of the sandbox
+    SandboxSpec spec = 1;
+}
+
+message CreateSandboxResponse {
+    Sandbox sandbox = 1;
+}
+
+message DeleteSandboxRequest {
+    // The id of the sandbox
+    string sanboxID = 1;
+}
+
+message DeleteSandboxResponse {
+}
+
+message SandboxStatusRequest {
+    // The id of the sandbox
+    string sanboxID = 1;
+}
+
+message SandboxStatusResponse {
+    SandboxStatus status = 1;
+}
+
+message ListSandboxRequest {
+}
+
+message ListSandboxResponse {
+    repeated Sanbox sanboxs = 1;
+}
+
+message SandboxStateRequest {
+    // The id of the sandbox
+    string sanboxID = 1;
+}
+
+message SandboxStateResponse {
+    SandboxState state = 1;
+}
+
+message SandboxMetricsRequest {
+    // The id of the sandbox
+    string sanboxID = 1;
+}
+
+message SandboxMetricsResponse {
+    SandboxMetrics metrics = 1;
+}
+
+message CreateContainerRequest {
+    ContainerSpec spec  = 1;
+}
+
+message CreateContainerResponse {
+    Container container = 1;
+}
+
+message StartContainerRequest {
+    // The id of the container
+    string containerID = 1;
+}
+
+message StartContainerResponse {
+}
+
+message StopContainerRequest {
+    // The id of the container
+    string containerID  = 1;
+    // Timeout in seconds to stop the container
+    uint32 timeout      = 2;
+}
+
+message StopContainerResponse {
+}
+
+message RemoveContainerRequest {
+    // The id of the container
+    string containerID = 1;
+}
+
+message RemoveContainerResponse {
+}
+
+message ListContainersRequest {
+
+}
+
+message ListContainersResponse {
+    repeated Container containers = 1;
+}
+
+message ContainerStatusRequest {
+    // The id of the container
+    string containerID = 1;
+}
+
+message ContainerStatusResponse {
+    ContainerStatus status = 1;
+}
+
+message ContainerMetricsRequst {
+    string containerID = 1;
+}
+
+message ContainerMetricsResponse {
+    ContainerMetrics metrics = 1;
+}
+
+message ExecInContainerRequest {
+    string containerID = 1;
+    repeated string cmd = 2;
+    bytes stdin = 3;
+}
+
+message ExecInContainerResponse {
+    bytes stdout = 1;
+    bytes stderr = 2;
+}
+
+message ListImagesRequest {
+}
+
+message ListImagesResponse {
+    repeated Image images = 1;
+}
+
+message ImageStatusRequest {
+    string image = 1;
+}
+
+message ImageStatusResponse {
+    Image image = 1;
+}
+
+message PullImageRequest {
+    string image = 1;
+    ImageSecret secret = 2;
+}
+
+message PullImageResponse {
+}
+
+message RemoveImageRequest {
+}
+
+message RemoveImageResponse {
+}
+
+message ImageMetricsRequest {
+    string image = 1;
+}
+
+message ImageMetricsResponse {
+    ImageMetrics imageMetrics = 1;
+}
+```
+
+Some type in the proto is already defined at [Container runtime interface/integration](https://github.com/kubernetes/kubernetes/pull/25899). We should decide how to integrate the types in [#25899](https://github.com/kubernetes/kubernetes/pull/25899) with gRPC services:
+
+* Auto-generate those types into proto by `go2idl`
```

Comment:
If we are going to use `grpc`, it seems easier to start from scratch and write raw proto files.

/cc @lavalamp, who's in charge of the k8s api machinery.